webots_ros2_driver/webots_ros2_driver/ros2_supervisor.py

```diff
@@ -0,0 +1,137 @@
+        self.create_subscription(String, 'remove_urdf_robot', self.__remove_urdf_robot_callback, qos_profile_services_default)
+
+    def __spawn_urdf_robot_callback(self, request, response):
+        robot = request.robot
+
+        robot_name = robot.name if robot.name else ''
+
+        if robot_name == '':
+            self.get_logger().info('Ros2Supervisor cannot import an unnamed URDF robot. Please specifiy it with name="" in the URDFSpawner object.')
+            response.success = False
+            return response
+        if robot_name in self.__urdf_robots_list:
+            self.get_logger().info('The URDF robot name "' + str(robot_name) + '" is already used by another robot! Please specifiy a unique name.')
+            response.success = False
+            return response
+
+        robot_translation = robot.translation if robot.translation else '0 0 0'
+        robot_rotation = robot.rotation if robot.rotation else '0 0 1 0'
+        normal = robot.normal if robot.normal else False
+        box_collision = robot.box_collision if robot.box_collision else False
+        init_pos = robot.init_pos if robot.init_pos else None
+
+        # Choose the conversion according to the input
+        if robot.urdf_path:
+            robot_string = convertUrdfFile(input=robot.urdf_path, robotName=robot_name, normal=normal,
+                                    boxCollision=box_collision, initTranslation=robot_translation, initRotation=robot_rotation,
+                                    initPos=init_pos)
+        elif robot.robot_description:
+            relative_path_prefix = robot.relative_path_prefix if robot.relative_path_prefix else None
+            robot_string = convertUrdfContent(input=robot.robot_description, robotName=robot_name, normal=normal,
+                                        boxCollision=box_collision, initTranslation=robot_translation, initRotation=robot_rotation,
+                                        initPos=init_pos, relativePathPrefix=relative_path_prefix)
+        else:
+            self.get_logger().info('Ros2Supervisor can not import a URDF file without a specified "urdf_path" or "robot_description" in the URDFSpawner object.')
+            response.success = False
+            return response
+
+        self.__insertion_robot_place.importMFNodeFromString(-1, robot_string)
+        self.get_logger().info('Ros2Supervisor has imported the URDF robot named "' + str(robot_name) + '".')
+        self.__urdf_robots_list.append(robot_name)
+        response.success = True
+        return response
+
+    def __remove_urdf_robot_callback(self, message):
+        robotName = message.data
+
+        if robotName in self.__urdf_robots_list:
+            robot_node = None
+
+            for id_node in range(self.__insertion_robot_place.getCount()):
+                node = self.__insertion_robot_place.getMFNode(id_node)
+                node_name_field = node.getField("name")
```

Comment:
We usually prefer single quotes:
```suggestion
                node_name_field = node.getField('name')
```